modules/tensor_mechanics/test/tests/beam/dynamic/tests

```diff
@@ -1,173 +1,287 @@
 [Tests]
+  design = "C0TimoshenkoBeam.md"
+  issues = "#10313"
   [./dyn_euler]
     type = 'Exodiff'
     input = 'dyn_euler_small.i'
     exodiff = 'dyn_euler_small_out.e'
+
+    requirement = "The dynamic system shall correctly predict the natural frequencies"
+                  " of an Euler-Bernoulli beam with consistent mass/inertia."
   [../]
   [./dyn_timoshenko]
     type = 'Exodiff'
     input = 'dyn_timoshenko_small.i'
     exodiff = 'dyn_timoshenko_small_out.e'
+
+    requirement = "The dynamic system shall correctly predict the natural frequencies"
+                  " of a Timoshenko beam with consistent mass/inertia."
   [../]
   [./dyn_euler_rayleigh_hht]
     type = 'Exodiff'
     input = 'dyn_euler_small_rayleigh_hht.i'
     exodiff = 'dyn_euler_small_rayleigh_hht_out.e'
+
+    requirement = "The dynamic system shall correctly predict the natural frequencies"
+                  " of an Euler-Bernoulli beam in the presence of Rayleigh damping"
+                  " and numerical damping induced by HHT time integration."
   [../]
   [./dyn_euler_added_mass]
     type = 'Exodiff'
     input = 'dyn_euler_small_added_mass.i'
     exodiff = 'dyn_euler_small_added_mass_out.e'
+
+    requirement = "The dynamic system shall correctly predict the natural frequencies"
+                  " of a massless Euler-Bernoulli beam with a nodal masses placed"
+                  " at the ends."
   [../]
   [./dyn_euler_added_mass_file]
     type = 'Exodiff'
     input = 'dyn_euler_small_added_mass_file.i'
     cli_args = "NodalKernels/x_inertial/nodal_mass_file=nodal_mass.csv"
     exodiff = 'dyn_euler_small_added_mass_out.e'
     prereq = 'dyn_euler_added_mass'
+
+    requirement = "The dynamic system shall correctly predict the natural frequencies"
+                  " of a massless Euler-Bernoulli beam with added nodal masses when"
+                  " the location and values of the masses are provided using a csv file."
   [../]
   [./dyn_euler_added_mass_gravity]
     type = 'Exodiff'
     input = 'dyn_euler_small_added_mass_gravity.i'
     cli_args = "NodalKernels/force_y2/mass=0.01899772"
     exodiff = 'dyn_euler_small_added_mass_out.e'
     prereq = 'dyn_euler_added_mass_file'
+
+    requirement = "The dynamic system shall correctly model the beam's response"
+                  " under gravitational force (proportional to nodal mass) is"
+                  " applied to the beam."
```

Comment:
The dynamic system shall correctly model the response of a beam modeled using beam elements when gravitational force (proportional to nodal mass) is applied to the beam.